src/Bundle/DependencyInjection/Driver/DriverProvider.php

```diff
@@ -13,41 +13,39 @@
 
 namespace Sylius\Bundle\ResourceBundle\DependencyInjection\Driver;
 
-use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrineODMDriver;
-use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrineORMDriver;
-use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrinePHPCRDriver;
 use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Exception\UnknownDriverException;
-use Sylius\Bundle\ResourceBundle\SyliusResourceBundle;
 use Sylius\Component\Resource\Metadata\MetadataInterface;
 use Webmozart\Assert\Assert;
 
 final class DriverProvider
 {
-    /** @var DriverInterface[] */
+    /** @var array<string, array<string, string>> */
```

Comment:
I'm not too familiar with this construction, but will this work?

```patch
--- @var array<string, array<string, string>>
+++ @var array<string, array<string, class-string<DriverInterface>>>
```

Will it remove need to specify
```php
       /** @var class-string|null $class */
        $class = self::$drivers[$type]['class'] ?? null;
```
and
```php
           /** @var DriverInterface $driver */
            $driver = new $class();
```
later in the file?

Reply:
Yes It works in this case, but the attribute php doc is here to define the generic type of a driver configuration. We should perhaps build a `DriverConfiguration` class which own this cconfiguration to better type it (I have to check if @loic425 already did this)